app/controllers/course_summaries_controller.rb

```diff
@@ -2,17 +2,50 @@ class CourseSummariesController < ApplicationController
   include CourseSummariesHelper
 
   before_action :authorize_only_for_admin,
-                except: [:populate]
+                except: [:populate, :index]
 
   layout 'assignment_content'
 
   def index
   end
 
   def populate
+    averages = {}
+    medians = {}
+    visible_assessment_totals = []
+    if current_user.admin?
+      visible_assessments = Assessment.all.order(id: :asc).pluck_to_hash(:id, :type, :short_identifier)
+      MarkingScheme.all.each do |m|
+        averages[m.name] = DescriptiveStatistics.mean(m.students_weighted_grades_array(current_user)).round(2)
+      end
+    else
+      visible_assessments = Assessment.where(is_hidden: false).order(id: :asc).pluck_to_hash(:id,
+                                                                                             :type,
+                                                                                             :short_identifier)
+    end
+    visible_assessments.each do |a|
+      if a[:type] == 'GradeEntryForm'
```

Comment:
After you make the previous change, you can use `a.is_a? GradeEntryForm` instead.